lib/pipeline/collection.rb

```diff
@@ -43,7 +43,7 @@ def each(per_page: nil)
     pages = nil
     while pages.nil? || page < pages
       response = read_page(page: page, per_page: per_page)
-      pages ||= response.is_a?(Array) || response[collection_name] ? 1 : response["pagination"]["pages"]
+      pages ||= response.is_a?(Array) || response[collection_name] || response["entries"] ? 1 : response["pagination"]["pages"]
```

Comment:
I'm not sure I understand the original logic or the additional change.

It seems like we are saying if the response is an array or contains the key mapping to the `collection_name`, then assume there is only one page.  This doesn't apply when there is the `entries` key, right?  I'm pretty sure when we set `entries`, we typically set the `pagination` as well.

Wouldn't it be safer to do something like...
```
pages ||= response.is_a?(Hash) && response.dig(:pagination, :pages) || 1
```

Reply:
I ran into something odd with `product_lines`, and you're right -- my change wasn't correct, but it did work, somehow! I'll try reverting and see what happens...

Reply:
https://github.com/PipelineDeals/pipeline_api/pull/21/files#diff-c9d1387711f78d40463ec42c409655e077d1535f88a174ef54684ef41784eb6bR63

Reply:
I like your suggestion...

Reply:
🎉 Great!